docs/source/examples/e2e_flow.rst

```diff
@@ -0,0 +1,519 @@
+using a ``batch_size=2`` and ``dtype=bfloat16``. With these settings the model
+should have a peak memory usage of ~16GB and total training time of around two hours for each epoch.
+We'll need to make some changes to the config to make sure our recipe can access the
+right checkpoints.
+
+Let's look for the right config for this use case by using the tune CLI.
+
+.. code-block:: bash
+
+    tune ls
+
+    RECIPE                                   CONFIG
+    full_finetune_single_device              llama2/7B_full_single_device
+                                             llama2/7B_full_single_device_low_memory
+                                             mistral/7B_full
+    full_finetune_distributed                llama2/7B_full
+                                             llama2/13B_full
+                                             mistral/7B_full
+    lora_finetune_single_device              llama2/7B_lora_single_device
+                                             llama2/7B_qlora_single_device
+                                             mistral/7B_lora
+    ...
+
+
+For this tutorial we'll use the ``llama2/7B_lora_single_device`` config.
+
+The config already points to the HF Checkpointer and the right checkpoint files.
+All we need to do is update the checkpoint directory for both the model and the
+tokenizer. Let's do this using the overrides in the tune CLI while starting training!
+
+
+.. code-block:: bash
+
+    tune run lora_finetune_single_device \
+    --config llama2/7B_lora_single_device \
+    checkpointer.checkpoint_dir=/tmp/Llama-2-7b-hf \
+    tokenizer.path=/tmp/Llama-2-7b-hf/tokenizer.model \
+    checkpointer.output_dir=/tmp/Llama-2-7b-hf
+
+
+Once training is complete, you'll see the following in the logs.
+
+.. code-block:: bash
+
+    [_checkpointer.py:473] Model checkpoint of size 9.98 GB saved to <checkpoint_dir>/hf_model_0001_0.pt
+
+    [_checkpointer.py:473] Model checkpoint of size 3.50 GB saved to <checkpoint_dir>/hf_model_0002_0.pt
+
+    [_checkpointer.py:484] Adapter checkpoint of size 0.01 GB saved to <checkpoint_dir>/adapter_0.pt
+
+
+The final trained weights are merged with the original model and split across two checkpoint files
+similar to the source checkpoints from the HF Hub
+(see the :ref:`LoRA Tutorial <lora_finetune_label>` for more details).
+In fact the keys will be identical between these checkpoints.
+We also have a third checkpoint file which is much smaller in size
+and contains the learnt LoRA adapter weights. For this tutorial, we'll only use the model
+checkpoints and not the adapter weights.
+
+|
+
+Run Evaluation using EleutherAI's Eval Harness
+----------------------------------------------
+
+We've fine-tuned a model. But how well does this model really do? Let's run some Evaluations!
+
+Instead of re-inventing the wheel on Evals, TorchTune integrates with
+`EleutherAI's evaluation harness <https://github.com/EleutherAI/lm-evaluation-harness>`_.
```

Comment:
Personally I would scrap the first half of this sentence. Or if anything add a sentence about eval harness and how great it is